test/functional/test_f_multi_keyring.py

```diff
@@ -0,0 +1,150 @@
+                    KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY,
+                    KeyringTraceFlag.WRAPPING_KEY_ENCRYPTED_DATA_KEY,
+                },
+            )
+        ],
+    ),
+]
+
+_MULTI_KEYRINGS = [
+    MultiKeyring(
+        generator=RawAESKeyring(
+            key_namespace=_PROVIDER_ID,
+            key_name=_KEY_ID,
+            wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+            wrapping_key=_WRAPPING_KEY_AES,
+        ),
+        children=[
+            RawRSAKeyring(
+                key_namespace=_PROVIDER_ID,
+                key_name=_KEY_ID,
+                wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+                wrapping_key=rsa.generate_private_key(public_exponent=65537, key_size=2048, backend=default_backend()),
+            ),
+            RawRSAKeyring(
+                key_namespace=_PROVIDER_ID,
+                key_name=_KEY_ID,
+                wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+                wrapping_key=rsa.generate_private_key(public_exponent=65537, key_size=2048, backend=default_backend()),
+            ),
+        ],
+    ),
+    MultiKeyring(
+        generator=RawRSAKeyring(
+            key_namespace=_PROVIDER_ID,
+            key_name=_KEY_ID,
+            wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+            wrapping_key=rsa.generate_private_key(public_exponent=65537, key_size=2048, backend=default_backend()),
+        )
+    ),
+    MultiKeyring(
+        children=[
+            RawRSAKeyring(
+                key_namespace=_PROVIDER_ID,
+                key_name=_KEY_ID,
+                wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+                wrapping_key=rsa.generate_private_key(public_exponent=65537, key_size=2048, backend=default_backend()),
+            ),
+            RawAESKeyring(
+                key_namespace=_PROVIDER_ID,
+                key_name=_KEY_ID,
+                wrapping_algorithm=WrappingAlgorithm.AES_128_GCM_IV12_TAG16_NO_PADDING,
+                wrapping_key=_WRAPPING_KEY_AES,
+            ),
+        ]
+    ),
+]
+
+
+@pytest.mark.parametrize("multi_keyring", _MULTI_KEYRINGS)
+@pytest.mark.parametrize("encryption_materials", _ENCRYPTION_MATERIALS)
+def test_multi_keyring_encryption_decryption(multi_keyring, encryption_materials):
+    # Call on_encrypt function for the keyring
+    encryption_materials = multi_keyring.on_encrypt(encryption_materials)
+
+    # Generate decryption materials
+    decryption_materials = DecryptionMaterials(
+        algorithm=Algorithm.AES_256_GCM_IV12_TAG16_HKDF_SHA384_ECDSA_P384, verification_key=b"ex_verification_key"
+    )
+
+    # Call on_decrypt function for the keyring
+    decryption_materials = multi_keyring.on_decrypt(
+        decryption_materials=decryption_materials, encrypted_data_keys=encryption_materials.encrypted_data_keys
+    )
+
+    if decryption_materials.data_encryption_key:
```

Comment:
This cycle test should carry an expectation that this is *always* true.

If there are scenarios to tests where this is *not* true, that should be tested in a different test function.